### PR TITLE
Feat/db refact + new context provider

### PR DESCRIPTION
**Pull request by Markkos89**
## Changes

- new context provider
- new way of fetching lessons/tracks with their status

### CODE REVIEW

File: prisma/seed.ts [hunk 1]
```diff
@@ -15,13 +15,20 @@ async function main() {
...
+  // const lesson1 = await prisma.lessons.create({
+  //   data: {
+  //     quizFileName: "lesson-4-quiz.json",
+  //   },
+  // });
+  // console.log({ lesson1 });
+  const completed2 = await prisma.completedQuizzes.create({
     data: {
-      quizFileName: "quiz-lesson-1.json",
+      userId: "cll3hcuim00001wujlay766tk",
```
Comment (wolovim) [inline prisma/seed.ts:26]: not sure what this is; anything sensitive about this ID? should it be passed in dynamically or via env var?

**Review by wolovim — APPROVED** [review]
auto-approving because this is in the dev branch of v1. i'm assuming these new patterns are closer to what exists or will exist in v2?

File: prisma/seed.ts (hunk 1 above)
Comment (Markkos89) [reply prisma/seed.ts:26]: > not sure what this is; anything sensitive about this ID? should it be passed in dynamically or via env var?

No no, this db query with prisma was used to populate my DB, and with the boolean test the `completed` state, instead or solving the Quiz :stuck_out_tongue_winking_eye:  :stuck_out_tongue_winking_eye:

**Comment by Markkos89** [comment]
> auto-approving because this is in the dev branch of v1. i'm assuming these new patterns are closer to what exists or will exist in v2? what isn't included in your PR text is "why" these changes are useful or necessary.

thanks for approving Marc.
I will merge it and soon another PR with details.

At this point, we should start testing the branch and fix every detail we decide to address :pray: 
I will appreciate it too much :pray: 

Regards!

**Pull request merged**